src/ssl.c

```diff
@@ -9116,12 +9116,20 @@ static int check_cert_key(DerBuffer* cert, DerBuffer* key, void* heap,
                 ) {
                 ret = wc_CryptoCb_RsaCheckPrivKey((RsaKey*)pkey,
                                                der->publicKey, der->pubKeySize);
+            #ifndef WOLF_CRYPTO_CB_ONLY_RSA
+                /* mark that fallback was used so the user can act accordingly */
+                ((RsaKey*)pkey)->cryptoCbSWFallback = 1;
```

Comment:
Don't you need to check the value of `ret` to determine if fallback will be used?